### ISSUE TITLE
[FEATURE] Ability to supply multiple filters to `find()` and `findOne()` methods

### ISSUE THREAD

**Issue by hakimio** [dxfrontier/cds-ts-repository#79]
### Description

Right now it seems it's only possible to supply a single instance of `Filter` to `find()` and `findOne()` method. It would be great if we could supply an array of `Filter` instances.

### Suggested solution

Following simple case assumes `and` relation for filters:
```ts
const results = await this.builder().find([filter1, filter2]).execute();
```

`or` operator can also be used:
```ts
const results = await this.builder().find([filter1, 'or', filter2]).execute();
```

Nested arrays can be used to group filters:
```ts
const results = await this.builder()
    .find([
        [filter1, 'or', filter2],
        'and',
        [filter3, 'or', filter4]
    ])
    .execute();
```

### Side note
You can find how this can be translated to query-by-example object on [capire](https://cap.cloud.sap/docs/node.js/cds-ql#where:~:text=Doing%20the%20same%20with%20object%20literals%20would%20look%20like%20that%3A).

**Comment by dragolea** [comment]
Hi,
Is this something that might help ?
Here you cam combine multiple filters in one filter 

```ts
Example 2 : Combination of 2...n filters

import { MyEntity } from 'LOCATION_OF_YOUR_ENTITY_TYPE';
import { Filter, BaseRepository } from '@dxfrontier/cds-ts-repository';

class MyRepository extends BaseRepository<MyEntity> {
  constructor() {
    super(MyEntity); // a CDS Typer entity type
  }

  public async aMethod() {
    const filterLike = new Filter<MyEntity>({
      field: 'customer_name',
      operator: 'LIKE',
      value: 'abs',
    });

    const filterBetween = new Filter<MyEntity>({
      field: 'stock',
      operator: 'BETWEEN',
      value1: 11,
      value2: 333,
    });

    const filterIn = new Filter<MyEntity>({
      field: 'ID',
      operator: 'IN',
      value: [201, 203, 207],
    });

    /*
    combinedLikeAndBetweenFilters translates to : 
    customer_name like 'abs' or stock between 11 and 333
    */
    const combinedLikeAndBetweenFilters = new Filter('OR', filterLike, filterBetween);

    /* 
    filters translates to : 
    (customer_name LIKE 'abs' OR stock BETWEEN 11 and 333) AND ID IN ('203', '201', '207')
    */
    const filters = new Filter('AND', combinedLikeAndBetweenFilters, filterIn);

    // Execute the query using the builder find
    const results = await this.builder().find(filters).getExpand('orders').execute();
    // OR
    // Execute the query using the .find
    const results2 = await this.find(filters);
  }
}
```

**Comment by hakimio** [comment]
While your suggested solution would solve the issue, I still think my proposed solution has much better DX. 
To be honest, I find it hard to follow the logic in your code.

**Comment by hakimio** [comment]
My solution was inspired by DevExtreme UI library: [DevExtreme DataSource filter](https://js.devexpress.com/Angular/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#filter).

**Comment by hakimio** [comment]
No rush.
Basically, to implement this you would have to convert proposed nested filter array to [sap _query-by-example_ nested object](https://cap.cloud.sap/docs/node.js/cds-ql#where:~:text=Doing%20the%20same%20with%20object%20literals%20would%20look%20like%20that%3A).
Shouldn't be too complicated.

**Comment by dragolea** [comment]
@hakimio Thanks for the input, this feature looks really nice.

**Comment by hakimio** [comment]
@dragolea thank you for implementing the feature 🙂

Where can I see what syntax is supported?
Is the following supported:
```ts
const filter = new Filter([
        [filter1, 'OR', filter2],
        'AND',
        [filter3, 'OR', filter4]
    ]);
```
Also, can you omit the `AND` operator?
```ts
const filter = new Filter([filter1, filter2]);
```

**Comment by dragolea** [comment]
You can see an example on [Filter](https://github.com/dxfrontier/cds-ts-repository?tab=readme-ov-file#filter)

Good point, currently my typing accepts this behaviour, I will test this case, I think it should support by default also this case, but I need to write a test and also to write it in readme if it's working or not.
```ts
const filter = new Filter([filter1, filter2]);
```

**Comment by dragolea** [comment]
@hakimio do you have a use case when you omit the `AND`, `OR`, I am thinking on a SQL level, but I cannot a uses case when you can filter over something without use `AND` or using logical `OR`.

If we omit `AND` or `OR` this should be forced by repository to an `AND` (which currently is not implemented) and :

```ts
const filter = new Filter([filter1, filter2]);
```
forced by repository will result in:

```sql
filter1 and filter2 
```
